src/steamship/base/response.py

```diff
@@ -28,6 +28,15 @@ class Config:
     def data(self):
         if self.error:
             raise self.error
+        if self.task is not None:
+            if self.task.state == TaskState.failed:
+                raise SteamshipError(
```

Comment:
here we could also check `task.statusMessage` which likely includes the reason for the failure. I believe there is also a `task.statusSuggestion` which might contain a possible remediation